### DISCUSSION TITLE
zero-size zero-length `etl::array` without compiler extension

### DISCUSSION THREAD

**Discussion by dhebbeker** [ETLCPP/etl#817] [Ideas]
Have you considered implementing a zero-length `etl::array` as `std::array` supports it? 

I am looking for a container which behaves like a C-style or `std::array` which has no data members (`sizeof(...)==0`) if the array has no elements. The compiler I use does not support the GCC extension of zero length arrays `T array[0]`.

I think the existing `etl::array` could be modified (using template specialization amongst others) to support this.

If you would agree on this feature I could provide a pull request. I expect the changes to be mostly boilerplate code.

**Comment by jwellbelove** [comment]
Yes, I would think that a simple specialisation should work.
Go ahead and do the pull request.

**Comment by dhebbeker** [comment]
I halt my attempt before beginning any implementation. At least for the time being. I realized that even empty classes have `sizeof(...) > 0` per language standard. It would be still possible to implement support for a zero-length array. But it would not satisfy my needs.
